app/assets/javascripts/organisation-autocomplete.js

```diff
@@ -5,16 +5,118 @@ window.GOVUKAdmin.Modules = window.GOVUKAdmin.Modules || {};
 
   Modules.OrganisationAutocomplete = function () {
     this.start = function ($element) {
-      accessibleAutocomplete.enhanceSelectElement({
-        selectElement: $element.find('#organisations')[0],
-        showAllValues: true,
-        autoselect: false,
-        dropdownArrow: dropdownArrow
-      });
-    };
 
-    function dropdownArrow() {
-      return '<span class="caret"></span>';
-    }
+      var autocompleteTemplateHtml = createAutocompleteTemplate();
+
+      $element
+        .append(addOrganisationButton())
+        .find('[selected]')
+        .each(function (index, option) {
+          if (index > 0) {
+            $(autocompleteTemplateHtml)
+              .insertAfter(lastSelectWrapper())
+              .find('option[value="' + $(option).attr('value') + '"]')
+              .attr('selected', 'selected')
+          }
+        });
+
+      enableAutocomplete();
+
+      function createAutocompleteTemplate() {
+        var $select = organisationSelects().first();
+        $select.removeAttr('multiple');
+        var $wrapper = wrapper($select);
+        var $template = $wrapper.clone();
+        $template.find('[selected]').removeAttr('selected');
+        return $template.prop('outerHTML');
+      }
+
+      function addOrganisationButton() {
```

Comment:
Why is the organisation button created via JS?

Reply:
Because we don't want it there when JS is disabled (progressive enhancement). We could alternatively have it present in the original HTML with `display: none` and unhide it in JS (similar to your suggestion for the template HTML).

Reply:
Unless you or @andrewgarner have strong opinions about using JS, I would prefer to use `HTML` via the `ERB` template.

Reply:
Fine by me, but I think we should pick one method and do that. So if you want this in HTML, I think _everything_ should be in HTML (including the template for creating new `select` elements, as you suggested elsewhere).

Reply:
Personally I think the HTML we have in the JavaScript module is correct. On the server we render only the HTML we need for the feature to work and then progressively we add additional behaviour and markup through the JavaScript module when needed and supported.

This is the same approach taken by the [accessible-autocomplete](https://github.com/alphagov/accessible-autocomplete) module and would likely be the approach taken if this behaviour were extracted into a shared GOV.UK component or if we decided to add multiple selection to [accessible-autocomplete](https://github.com/alphagov/accessible-autocomplete).

Reply:
Yeah, it's hard to say. Arguably the accessible autocomplete adds HTML through JavaScript, because that's how React and JSX is meant to be used, whereas it could be said that jQuery is intended mainly to manipulate existing DOM elements rather than create new ones.

If you wanted to be _really_ purist about it, we could share our HTML templates between JS and Ruby using [mustache](https://github.com/alphagov/shared_mustachehttps://github.com/alphagov/shared_mustache). That would let us define an HTML template once, but without having "useless" HTML floating around on the page, or declaring HTML markup in JS.

Reply:
We should decide what capabilities we gain or loose instead of thinking in terms of whether HTML should or should not belong in the JavaScript module.

As it stands now:

- The additional elements are only required when we progressively add the behaviour.
- We're not sharing the templates with the server so we're not duplicating markup.

Reply:
Let's not block the PR for this reason. Let's keep the HTML in the JS for now.